hathor/p2p/sync_v2/mempool.py

```diff
@@ -52,21 +54,28 @@ def is_running(self) -> bool:
         """Whether the sync-mempool is currently running."""
         return self._is_running
 
-    def run(self) -> None:
+    def run(self) -> Deferred[None]:
         """Starts _run in, won't start again if already running."""
         if self.is_running():
             self.log.warn('already started')
-            return
+            return succeed(None)
```

Comment:
**question:** shouldn't we return `self.deferred` here?

I got that the deferred is only resolved when the execution is complete, but here it's still running.

Reply:
I agree that returning the current deferred is better than creating a new one that is resolved. I just made the change. Thanks!